flink-connectors/flink-connector-kafka-base/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumerBase.java

```diff
@@ -480,7 +477,7 @@ public void open(Configuration configuration) throws Exception {
 			}
 
 			for (Map.Entry<KafkaTopicPartition, Long> restoredStateEntry : restoredState.entrySet()) {
-				if (!restoredFromOldState) {
+				if (discoveryIntervalMillis != PARTITION_DISCOVERY_DISABLED) {
```

Comment:
This, therefore, is a bit confusing.

If we decide to change it so that whether or not partition discovery is enabled is determined at the very beginning and cannot be changed from them on (even on restarts), then there shouldn't be a case to check `discoveryIntervalMillis != PARTITION_DISCOVERY_DISABLED` here.